web_src/js/utils/image.js

```diff
@@ -0,0 +1,44 @@
+export async function pngChunks(blob) {
+  const uint8arr = new Uint8Array(await blob.arrayBuffer());
+  const chunks = [];
+  if (uint8arr.length < 12) return chunks;
+  const view = new DataView(uint8arr.buffer);
+  if (view.getBigUint64(0) !== 9894494448401390090n) return chunks;
+
+  const decoder = new TextDecoder();
+  let index = 8;
+  while (index < uint8arr.length) {
+    const len = view.getUint32(index);
+    chunks.push({
+      name: decoder.decode(uint8arr.slice(index + 4, index + 8)),
+      data: uint8arr.slice(index + 8, index + 8 + len),
+    });
+    index += len + 12;
+  }
+
+  return chunks;
+}
+
+export async function pngInfo(blob) {
+  let width = 0;
+  let dppx = 1;
+  if (blob.type !== 'image/png') return {width, dppx};
```

Comment:
A nit, TBH, I never saw a function returns a valid "struct" when it can't parse something, the literal meaning of "{width: 0}" for a non-png image seems confusing.


Maybe it should throw an exception or return null?

Reply:
Yeah this is something I thought about changing as well. Previously it was called `imageInfo` and in such a case it made sense to fall back to default values. I will at least move the type check out of the function, not sure about throwing, thought.

Reply:
I renamed it back to `imageInfo` as I could see it being extended with other types like webp oder jpg later.

I disagree about returning null or throwing, it would complicate the call site of the function unecessary, and right now because we always return a object it can stay compact. Also I think it's good to have a static return structure which will make it a breeze to convert into Typescript later.

Reply:
> I disagree about returning null or throwing, it would complicate the call site of the function unecessary, and right now because we always return a object the call site can stay compact and tidy.

Well, I have never seen a widely-used framework does so. (not a blocker, just my thought)

Reply:
One example would be various React APIs, it's very commonplace to return objects that are meant to be destructured. For example: https://docs.dndkit.com/api-documentation/draggable/usedraggable.

The object is just there to represent multiple named return values, something JS ist not capable of.

Reply:
That's different. I guess `useDraggable` never parses something, and it never fails.

Reply:
These `use*`functions can also fail, but the point is they would throw instead of returning a non-object value (which would lead to a destructuring error at the caller).

I checked what is needed to return null in our case and its at least +6 lines for no real benefit.